strax/context.py

```diff
@@ -781,36 +797,44 @@ def estimate_run_start(self, run_id, targets=None):
             for t in strax.to_str_tuple(targets):
                 try:
                     t0 = self.get_meta(run_id, t)['chunks'][0]['start']
-                    return (int(t0) // int(1e9)) * int(1e9)
+                    t0 = (int(t0) // int(1e9)) * int(1e9)
+
+                    t1 = self.get_meta(run_id, t)['chunks'][-1]['end']
+                    t1 = (int(t1) // int(1e9)) * int(1e9)
+                    return t0, t1
                 except strax.DataNotAvailable:
                     pass
         warnings.warn(
-            "Could not estimate run start time from "
-            "run metadata: assuming it is 0",
+            "Could not estimate run start and end time from "
+            "run metadata: assuming it is 0 and inf",
             UserWarning)
-        return 0
-
+        return 0, float('inf')
 
-    def to_absolute_time_range(self, run_id, targets, time_range=None,
-                               seconds_range=None, time_within=None):
+    def to_absolute_time_range(self, run_id, targets=None, time_range=None,
+                               seconds_range=None, time_within=None,
+                               full_range=None):
         """Return (start, stop) time in ns since unix epoch corresponding
         to time range.
 
+        :param run_id: run id to get
         :param time_range: (start, stop) time in ns since unix epoch.
         Will be returned without modification
         :param targets: data types. Used only if run metadata is unavailable,
         so run start time has to be estimated from data.
         :param seconds_range: (start, stop) seconds since start of run
         :param time_within: row of strax data (e.g. eent)
+        :param full_range: If True returns full time_range of the run.
         """
-        if ((time_range is None)
-                + (seconds_range is None)
-                + (time_within is None)
-                < 2):
+        
+        selection = ((time_range is None) +
+                     (seconds_range is None) +
+                     (time_within is None) +
+                     (full_range is None))
+        if selection < 2:
```

Comment:
@WenzDaniel , could you make it a bool and update the Runtime error?
As in : 
full_range=False # as kwarg
full_range==False # in selection comparison

and 
 " time_range, seconds_range, time_within or full_range"

Reply:
I think a bool is cleaner for an option but it's not too relevant

Reply:
I corrected the raise message. About the other thing, first I thought so too. But I think Jelle did this intentionally to allow arrays as input. Otherwise things like 

`if np.array([start, stop]) :`

would give an ambiguity error and one would have to change it in something like:

`if np.any(selection) :`